db.js

```diff
@@ -148,15 +159,19 @@ exports.init = function (sbot, config) {
     const guard = guardAgainstDuplicateLogs('add()')
     if (guard) return cb(guard)
 
-    stateFeedsReady.promise.then(() => {
-      try {
-        state = validate.append(state, hmac_key, msg)
-        if (state.error) return cb(state.error)
-        rawAdd(msg, true, cb)
-      } catch (ex) {
-        return cb(ex)
+    onceWhen(
```

Comment:
I have a little trouble wrapping my head around how this functions when migrate is not enabled? Can you walk me through how it works in onceWhen? Like is there a fast path it will take then?

Reply:
I think this code is not related to migrate in any way, when migrate is turned off, this `onceWhen` will just listen to the `stateFeedsReady` Obv (it used to be a promise, but we need to bring it back to `false` (or "migrating") sometimes, so I made it an Obv), and when that Obv finally satisfies the second argument `ready => ready === true`, then the cb is called and it will never be called again (because it's "once").

Reply:
I ran a test locally, mostly just so I understand how this works so mostly the calls just runs through the Obz. Then overhead doesn't seem quite minimal. I noticed there was no test of the performance of adding a bunch of messages so I tried that locally. Will do a separate PR for that. The result was the same though for this and master.